src/dfx-core/src/error/canister_id_store.rs

```diff
@@ -32,6 +35,9 @@ pub enum CanisterIdStoreError {
         canister_id: String,
         cause: UnifiedIoError,
     },
+
+    #[error(transparent)]
+    PullIdsError(#[from] DfxConfigError),
```

Comment:
A transparent error just passes through the underlying error, without an assumption that it is about some other specific thing
```suggestion
    DfxConfigError(#[from] DfxConfigError),
```